### PR TITLE
feat(wzn): implement and optimize evaluation for SRM and GRM trainers

### PR DESCRIPTION
**Pull request by zunian-wan**
- Implement `generate` method in `GenerativeRewardModelVL`.
- Refactor `GRMTrainerVL.evaluate`.
-  Implement `SRMTrainerAL.evaluate` and `SRMTrainerVL.evaluate` to calculate and log accuracy and reward metrics (chosen/rejected).
- Update `GRMDataset` to correctly strip assistant messages during evaluation tokenization.
- Add `generate_max_len`, `eval_data`, and `eval_steps` arguments to training scripts and shell examples.

### CODE REVIEW

File: lightrft/models/grm_vl.py [hunk 1]
```diff
@@ -167,6 +167,69 @@ def forward(
...
+        if kwargs.get("max_new_tokens", None):
+            generate_args["max_new_tokens"] = kwargs.get("max_new_tokens")
+        if kwargs.get("max_length", None):
+            generate_args["max_length"] = kwargs.get("max_length")
+
+        for k, v in kwargs.items():
+            if k not in generate_args:
+                generate_args[k] = v
+
+        unwarp_model = self.model
```
Comment (PaParaZz1) [inline lightrft/models/grm_vl.py]: 直接 self.model.generate 呗

File: lightrft/trainer/grm_trainer_vl.py [hunk 2]
```diff
@@ -289,34 +291,76 @@ def evaluate(self, args, eval_dataloader, steps: int = 0) -> None:
...
-            if self._wandb is not None and self.strategy.is_rank_0():
-                logs = {"train/%s" % k: v for k, v in {**logs_dict, "global_step": steps}.items()}
-                self._wandb.log(logs)
-            # TensorBoard
-            elif self._tensorboard is not None and self.strategy.is_rank_0():
-                for k, v in logs_dict.items():
-                    self._tensorboard.add_scalar(f"train/{k}", v, steps)
+            self.strategy.print(f"Evaluation generations written to {output_file}")
 
-        self.model.train()  # reset model state
```
Comment (PaParaZz1) [inline lightrft/trainer/grm_trainer_vl.py]: 这注释为啥删了

File: lightrft/trainer/grm_trainer_vl.py [hunk 3]
```diff
@@ -289,34 +291,76 @@ def evaluate(self, args, eval_dataloader, steps: int = 0) -> None:
...
-                "eval_loss": sum(total_loss).item() / len(total_loss),
-            }
-            for k in logs_dict.keys():
-                logs_dict[k] = self.strategy.all_reduce(logs_dict[k])
+        if self.strategy.is_rank_0():
+            # Write JSON file
+            with open(output_file, 'w') as f:
+                json.dump(all_eval_records, f, indent=4, ensure_ascii=False)
 
-            # wandb
```
Comment (PaParaZz1) [inline lightrft/trainer/grm_trainer_vl.py]: wandb和tb为啥删了，应该要把一些eval结果打进去的呀

File: lightrft/trainer/grm_trainer_vl.py [hunk 4]
```diff
@@ -244,7 +246,6 @@ def save_logs_and_checkpoints(
                 self.model, args.ckpt_path, tag, args.max_ckpt_num, args.max_ckpt_mem, client_states
             )
 
-    # TODO: eval for vl grm
     def evaluate(self, args, eval_dataloader, steps: int = 0) -> None:
         """
         Evaluate the model on the provided dataloader.
```
Comment (PaParaZz1) [inline lightrft/trainer/grm_trainer_vl.py]: 这里的注释也扩充下吧

**Review by PaParaZz1 — CHANGES_REQUESTED** [review]

File: lightrft/trainer/grm_trainer_vl.py (hunk 2 above)
Comment (zunian-wan) [reply lightrft/trainer/grm_trainer_vl.py]: 误删，加回去了

File: lightrft/trainer/grm_trainer_vl.py (hunk 3 above)
Comment (zunian-wan) [reply lightrft/trainer/grm_trainer_vl.py]: 已修复，wandb 和 tb 的 logging
